core/services/headtracker/head_listener.go

```diff
@@ -187,7 +186,9 @@ func (hl *HeadListener) subscribeToHead() error {
 
 	hl.headers = make(chan *eth.Head)
 
-	sub, err := hl.ethClient.SubscribeNewHead(context.Background(), hl.headers)
+	ctx, cancel := utils.ContextFromChan(hl.chStop)
```

Comment:
we have such the context created in `ListenForNewHeads`, can't we just pass that context all the way down to `subscribeToHead`?